learningequality/studio
# Updates to assessment items backend

## Description

* Adds an index to assessment_id field on assessment items to improve query performance on updates keyed by assessment_id
* Adds tests to the assessment item viewset and fixes behaviour
* Updates the valuesviewset and sync machinery to handle combined indexes
* Applies combined index behaviour to assessment item endpoint (contentnode, assessment_id)
* Updates the frontend assessment item resource to use a combined index of contentnode and assessment_id
* Upgrades DexieJS to version 3 to allow for easier schema migrations (adding indexes no longer requires explicit upgrade logic)
* Fixes a hitherto unreported bug whereby the leader election success callback would attempt to run a sync before we had called open on the database

#### Issue Addressed (if applicable)

Fixes #2252
Fixes #1995
Fixes #1843


## Steps to Test

You will need to clear your IndexedDB as this PR updates the primary key of the assessment item table.

* Create an assessment item
* Edit an assessment item
* Delete an assessment item

Verify that all these operations work as expected.
